lib/capistrano/opscomplete/ruby.rake

```diff
@@ -106,12 +106,52 @@ namespace :opscomplete do
           raise Capistrano::ValidationError,
                 "#{host}: Configured Ruby version is neither installed nor installable using ruby-install."
         end
-        execute(:rbenv, :global, "'#{app_ruby_version}'") unless capture(:rbenv, :global) == app_ruby_version
+        unless capture(:rbenv, :global) == app_ruby_version
+          set :ruby_modified, true
+          execute(:rbenv, :global, "'#{app_ruby_version}'")
+        end
       end
       invoke('opscomplete:ruby:install_rubygems')
       invoke('opscomplete:ruby:install_bundler')
       invoke('opscomplete:ruby:install_geordi')
       invoke('opscomplete:ruby:rehash') if fetch(:rbenv_needs_rehash, false)
     end
+
+    desc 'resets the global ruby version and gems to Gemfile and .ruby-version in current_path.'
+    task :reset do
+      on roles fetch(:rbenv_roles, :all) do |host|
+        within current_path do
+          current_ruby_version_file_path = current_path.join('.ruby-version').to_s
+          if test("[ -f #{current_ruby_version_file_path} ]")
+            execute(:rbenv, :global, capture(:cat, current_ruby_version_file_path))
+          else
+            raise Capistrano::ValidationError,
+                  "#{host}: Missing .ruby version in #{current_path}. Won't set a new global version."
+          end
+          if test("[ -f '#{current_path}/.bundle/config' ]")
+            debug("#{host}: Found #{current_path}/.bundle/config, running bundle pristine.")
+            set :bundle_gemfile, -> { current_path.join('Gemfile') }
+            execute(:bundle, :pristine)
+          else
+            raise Capistrano::ValidationError,
+              "Unable to find #{current_path}/.bundle/config, won't run bundle pristine."
+          end
+        end
+      end
+    end
+
+    desc 'Set the old ruby version before the change and invoke bundle pristine.'
+    task :broken_gems_warning do
+      on roles fetch(:rbenv_roles, :all) do |host|
+        if fetch(:ruby_modified, false)
+          warn("Deploy failed and the ruby version has been modified in this deploy.")
+          warn("If this was a minor ruby version upgrade your running application may run into issues with native gem extensions.")
+          warn("If your deploy failed deploy:symlink:release you may run bundle exec cap #{fetch(:stage)} opscomplete:ruby:reset.")
```

Comment:
I think it would be better to read if the command to execute is enclosed in e.g. *`*.

```
          warn("If your deploy failed deploy:symlink:release you may run `bundle exec cap #{fetch(:stage)} opscomplete:ruby:reset`.")
```